src/rust/ensogl/lib/core/src/gui/component.rs

```diff
@@ -198,22 +253,45 @@ impl<T> display::Object for ShapeView<T> {
 #[derive(CloneRef,Derivative,Debug,Shrinkwrap)]
 #[derivative(Clone(bound=""))]
 #[allow(missing_docs)]
-pub struct Animation<T> {
+pub struct Animation<T:Animatable> {
     #[shrinkwrap(main_field)]
-    pub simulator : DynSimulator<T>,
+    pub simulator : DynSimulator<T::AnimationSpaceRepr>,
     pub value     : frp::Stream<T>,
 }
 
-impl<T:inertia::Value> Animation<T> {
+#[allow(missing_docs)]
+impl<T:Animatable+Debug+Default+Clone+'static> Animation<T> {
     /// Constructor.
     pub fn new(network:&frp::Network) -> Self {
         frp::extend! { network
             def target = source::<T>();
         }
-        let simulator = DynSimulator::<T>::new(Box::new(f!((t) target.emit(t))));
+        let simulator = DynSimulator::<T::AnimationSpaceRepr>::new(Box::new(f!((t) {
+             target.emit(AnimationLinearSpace{value:t}.into())
```

Comment:
Change it pls to `t.to_animation_linear_space()` - this can be a method generated automatically which uses `into` under the hood. Lets apply it to other places below as well :) Manual construction of wrappers everywhere adds a lot of boilerplate

Reply:
I added a `from_animation_space` function that does that.